deploy/compose/osie/lastmile.sh

```diff
@@ -47,44 +48,54 @@ hook_rename_files() {
 	mv "${src_initrd}" "${dest_dir}/initramfs-x86_64"
 }
 
-# main runs the functions in order to download, extract, and move helper scripts
-main() {
+# process_osie_files processes the OSIE files and moves them to the correct location
+process_osie_file() {
 	local url="$1"
-	local extract_dir="$2"
-	local source_dir="$3"
-	local dest_dir="$4"
-	local use_hook="$5"
-
-	local filename
-	if [ "${use_hook}" == "true" ]; then
-		filename="osie-hook"
-	else
-		filename="osie"
-	fi
+	local filename="$2"
+	local extract_dir="$3"
+	local source_dir="$4"
+	local dest_dir="$5"
+	local use_hook="$6"
 
 	if [ ! -f "${extract_dir}"/"${filename}".tar.gz ]; then
-		echo "downloading osie..."
+		echo "downloading osie(s)..."
```

Comment:
this is still only downloading one file at a time, so singular makes sense to keep.

Reply:
updated